js/adapt-assessmentArticleModel.js

```diff
@@ -460,14 +469,21 @@ define([
             Backbone.history.navigate("#/id/"+Adapt.location._currentId, { replace:true, trigger: true });
         },
 
-        _resetQuestions: function() {
+        _resetQuestions: function(callback) {
             var assessmentConfig = this.getConfig();
-            var questionComponents = this._currentQuestionComponents;
+            var syncIterations = 1; // number of synchronous iterations to perform
+            var i = 0, qs = this._currentQuestionComponents, len = qs.length;
 
-            for (var i = 0, l = questionComponents.length; i < l; i++) {
-                var question = questionComponents[i];
-                question.reset(assessmentConfig._questions._resetType, true);
+            function step() {
+                for (var j=0, count=Math.min(syncIterations, len-i); j < count; i++, j++) {
+                    var question = qs[i];
+                    question.reset(assessmentConfig._questions._resetType, true);
+                }
+		console.log(i==len?"all reset":"reset question step");
```

Comment:
Can you please remove the console.log.
We might add a develop level log output once the logging system is implemented.